File: files/300-399/305/CH-305.R
library(tidyverse)
library(readxl)

path = "files/300-399/305/CH-305 Advanced Calculation.xlsx"
input = read_excel(path, range = "B2:D19")
test  = read_excel(path, range = "H2:I5") 

result = input %>%
  summarise(Index = diff(range(Sales)/mean(Sales[!Sales %in% range(Sales)])), .by = Product)

all.equal(result, test)
# wrong answer for product A

File: files/300-399/305/CH-305.py
import pandas as pd

path = "300-399/305/CH-305 Advanced Calculation.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=17)
test = pd.read_excel(path, usecols="H:I", skiprows=1, nrows=4)

result = input.groupby('Product')['Sales'].apply(
    lambda x: (x.max() - x.min()) / x.drop([x.idxmin(), x.idxmax()]).mean()
)

print(result)
# Different result for Product A